src/mesh_refiner.cpp

```diff
@@ -184,6 +184,14 @@ std::size_t Mesh_Refiner::mark_for_refinement (
       _cell_marker->set_value(index, true);
       marked_count++;
     }
+    if (Mesh_Refiner::_mesh->num_cells()+5*marked_count > Mesh_Refiner::max_elements)
```

Comment:
This only occurs when `mark_for_refinement` is called.  The mesh marked out by `mark_for_refinement` is only used if it provides a mesh that has at most `max_elements` cells; otherwise, the mesh refinement algorithm goes into `mark_for_refinement_with_target_size`, which is more conservative.

Does this if statement ever evaluate to `true`?